File: test-support/src/main/java/com/dbx/agent/test/JdbcConnectedAgentTest.java
package com.dbx.agent.test;

import com.dbx.agent.DatabaseAgent;

import java.util.function.Consumer;

public abstract class JdbcConnectedAgentTest {
    protected abstract DatabaseAgent createConnectedAgent(String databaseName);

    protected void withAgent(String databaseName, Consumer<DatabaseAgent> block) {
        DatabaseAgent agent = createConnectedAgent(databaseName);
        try {
            block.accept(agent);
        } finally {
            agent.disconnect();
        }
    }
}

File: test-support/src/main/kotlin/com/dbx/agent/test/JdbcConnectedAgentTest.kt
package com.dbx.agent.test

import com.dbx.agent.DatabaseAgent

abstract class JdbcConnectedAgentTest {
    protected abstract fun createConnectedAgent(databaseName: String): DatabaseAgent

    protected fun withAgent(databaseName: String, block: (DatabaseAgent) -> Unit) {
        val agent = createConnectedAgent(databaseName)
        try {
            block(agent)
        } finally {
            agent.disconnect()
        }
    }
}
